src/indexer/merger.rs

```
@@ -920,19 +920,26 @@ impl IndexMerger {
             .iter()
             .map(|reader| reader.get_store_reader())
             .collect::<Result<_, _>>()?;
+        let mut document_iterators: Vec<_> = store_readers
+            .iter()
+            .enumerate()
+            .map(|(i, store)| store.iter_raw(self.readers[i].delete_bitset()))
+            .collect();
         if let Some(doc_id_mapping) = doc_id_mapping {
             for (old_doc_id, reader_with_ordinal) in doc_id_mapping {
-                let store_reader = &store_readers[reader_with_ordinal.ordinal as usize];
-                let raw_doc = store_reader.get_raw(*old_doc_id)?;
+                let store_reader = &mut document_iterators[reader_with_ordinal.ordinal as usize];
+                let raw_doc = store_reader.next().expect(&format!(
```

can we return an error here? (the error message is great.)